### ISSUE TITLE
Model.set wiping out other properties in the model

### ISSUE THREAD

**Issue by vijayst** [wearevolt/mobx-model#21]
After an update API action, I am using `Model.set({ modelJson: attributes })` as follows.

```
update(attributes = {}) {
    API.requestHeaders[HEADER_SESSION_TOKEN] = authProvider.fetchToken();
    return API.request({
      method: "put",
      data: { resource: [attributes] },
      endpoint: this.urlRoot,
      onSuccess: (response) => {
        console.log(this.detail);
        this.set({ modelJson: attributes });
        console.log(this.detail);
      }
    });
  }
```

This is wiping out all the other properties.

In attributes, I passed in "id" and "name". The console.log before and after the set is shown below.

![screen shot 2016-10-11 at 3 49 56 pm](https://cloud.githubusercontent.com/assets/3457565/19267187/09edfaf2-8fcb-11e6-99d8-a817d944e988.png)

**Comment by vijayst** [comment]
It will be nice to add another method: `Model.setDelta` which will only update the Model with the properties sent in the setDelta method.

```
BaseModel.addAction('update', function(attributes = {}) {
  return API.request({
    method: 'put',
    data: attributes,
    endpoint: `${this.urlRoot}/${this.id}`,
    onSuccess: (response) => {
      this.setDelta({
        modelJson: attributes,
      });
    }
  });
});
```

**Comment by ikido** [comment]
If you want to update certain models you should use `model.set()`, same method but on model instance. 

```
const id = 1;
const model = Mogel.get(id);
model.set({ modelJson: attributes })
```

if you are adding action to an instance you can just use `this.set()`, it will call this method on model instance

**Comment by vijayst** [comment]
`this.set({ modelJson: attributes })` replaces the model with whatever is there in attributes. Prefer it to retain other attributes.

**Comment by ikido** [comment]
You are right. What about a flag to modify the befaviour, such as `model.set({ modelJson: attributes, reset: false })`? I don't think it's a good idea to add more methods with similar functionality

**Comment by vijayst** [comment]
Thanks @ikido  
Having a `reset: false` should work.
